File: ObjC/Length/Centimeter.m
//
//  Centimeter.m
//  Pods
//
//  Created by materik on 15/02/16.
//
//

#import "Centimeter.h"

#import "Foot.h"
#import "Inch.h"
#import "Meter.h"

@implementation Centimeter

+ (NSString *)unit {
    return @"cm";
}

+ (NSString *)unitLong {
    return @"centimeters";
}

- (Centimeter *)convertToCentimeter {
    return [Centimeter value:self.value];
}

- (Foot *)convertToFoot {
    return [Foot value:self.value * 0.032808399f];
}

- (Inch *)convertToInch {
    return [Inch value:self.value * 0.3937007874f];
}

- (Meter *)convertToMeter {
    return [Meter value:self.value * 0.01f];
}

@end

File: Swift/Length/Centimeter.swift
//
//  Centimeter.swift
//  Pods
//
//  Created by materik on 16/02/16.
//
//

import Foundation

public class Centimeter: Unit, Length {
    
    override var unit: String { return "cm" }
    
    public func convertToCentimeter() -> Centimeter {
        return Centimeter(self.value)
    }
    
    public func convertToMeter() -> Meter {
        return Meter(self.value / 100.0)
    }
    
}
